### ISSUE TITLE
Shader Modifiers

### ISSUE THREAD

**Issue by d-ronnqvist** [d-ronnqvist/SCNBook-code#3]
Shader Modifiers is one of the most convenient rendering features of Scene Kit (in my opinion). Unfortunately it didn't make the cut for the first release, but is a top priority for the first update.

**Comment by donhuff** [comment]
Love chapters on GLSL, the rendering pipeline and custom shaders. A shader modifier chapter would be awesome.

**Comment by gsabran** [comment]
Yes, it'd be great! Especially since Apple documentation is sometime hard to get to work. Some examples with Metal would also be great here!
